src/main/drivers/compass/compass_ak8963.c

```diff
@@ -43,74 +46,72 @@
 #include "drivers/accgyro/accgyro_spi_mpu6500.h"
 #include "drivers/accgyro/accgyro_spi_mpu9250.h"
 #include "drivers/compass/compass_ak8963.h"
-#include "drivers/compass/compass_spi_ak8963.h"
 
 static float magGain[3] = { 1.0f, 1.0f, 1.0f };
 
-#if defined(MPU6500_SPI_INSTANCE) || defined(MPU9250_SPI_INSTANCE)
-static busDevice_t *bus = NULL;
+#if defined(USE_MAG_AK8963) && (defined(USE_GYRO_SPI_MPU6500) || defined(USE_GYRO_SPI_MPU9250))
 
-static bool spiWriteRegisterDelay(const busDevice_t *bus, uint8_t reg, uint8_t data)
+static bool ak8963SpiWriteRegisterDelay(const busDevice_t *bus, uint8_t reg, uint8_t data)
 {
     spiBusWriteRegister(bus, reg, data);
     delayMicroseconds(10);
     return true;
 }
 
-typedef struct queuedReadState_s {
-    bool waiting;
-    uint8_t len;
-    uint32_t readStartedAt; // time read was queued in micros.
-} queuedReadState_t;
-
-typedef enum {
-    CHECK_STATUS = 0,
-    WAITING_FOR_STATUS,
-    WAITING_FOR_DATA
-} ak8963ReadState_e;
-
-static queuedReadState_t queuedRead = { false, 0, 0};
-
-static bool ak8963SensorRead(uint8_t addr_, uint8_t reg_, uint8_t len_, uint8_t *buf)
+static bool ak8963SlaveReadRegisterBuffer(const busDevice_t *slavedev, uint8_t reg, uint8_t *buf, uint8_t len)
```

Comment:
I think we are missing here the SPI bus and the CS pin parameters to make it work. Likely the busDevice union need to be extended for the slave case?